tmt/hardware.py

```diff
@@ -731,12 +731,50 @@ def _cast_number(raw_value: Any) -> int:
             name,
             raw_value,
             as_quantity=False,
-            as_cast=_cast_number,
+            as_cast=_cast_int,
             original_constraint=original_constraint,
             allowed_operators=allowed_operators
             )
 
 
+class NumberConstraint(Constraint[float]):
+    """ A constraint representing a dimension-less float number """
+
+    @classmethod
+    def from_specification(
+            cls: type[T],
+            name: str,
+            raw_value: str,
+            original_constraint: Optional['Constraint[Any]'] = None,
+            allowed_operators: Optional[list[Operator]] = None,
+            default_unit: Optional[Any] = 'MHz'
```

Comment:
Most likely `None` would be better, `MHz` would make sense for just some of the requirements. Note that the original `NumberConstraint` did not have a default unit at all, it's just a dimensionless number most of the time. This one wouldn't be that different, and only *some* would need to specify this field.